### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by roman4hik**

### CODE REVIEW

File: steepshot_io/graph/urls.py [hunk 1]
```diff
@@ -3,24 +3,24 @@
...
-    url(r'^posts/count/weekly$', CountPostWeekly.as_view(), name='posts_count_weekly'),
-    url(r'^posts/average/weekly$', PostsAverageWeekly.as_view(), name='posts_average_weekly'),
-    url(r'^posts/average/author$', PostsAverageAuthor.as_view(), name='posts_average_author'),
-    url(r'^count/votes/weekly$', CountVotesWeekly.as_view(), name='count_votes_weekly'),
-    url(r'^votes/average/weekly$', AverageVotesWeekly.as_view(), name='votes_average_weekly'),
-    url(r'^count/comments/weekly$', CountCommentsWeekly.as_view(), name='count_comments_weekly'),
-    url(r'^count/users/sessions$', CountUsersSessions.as_view(), name='users_count_session'),
-    url(r'^posts/fee/weekly$', PostsFeeWeekly.as_view(), name='posts_fee_weekly'),
-    url(r'^posts/fee/users$', PostsFeeWeekly.as_view(), name='posts_fee_users'),
+    url(r'^posts/count/monthly', GetPostsCountMonthly.as_view(), name='count_posts'),
```
Comment (alekseitylindus) [inline steepshot_io/graph/urls.py]: Why did you remove '$' from the end of urlpattern? 'posts/count/dailyhahaha' is valid too in this case

File: steepshot_io/graph/views.py [hunk 2]
```diff
@@ -39,110 +49,86 @@ def group_steem_golos(self, list_steem, list_golos):
...
         return res
 
-    def get_posts_count_monthly(self, platform=None):
-
+    def _get_data(self, platform=None, reverse=True, data_x=None, data_y=None, name_url=None):
         if platform == 'steem':
-            res = requests.get(settings.REQUESTS_URL_STEEM['posts_count_monthly']).json()
-            res.reverse()
+            try:
+                res = requests.get(settings.REQUESTS_URL_STEEM[name_url]).json()
```
Comment (AleksandrLeonov) [inline steepshot_io/graph/views.py]: Please move this code to separate method because you use the same code for steem and golos platform.

File: steepshot_io/graph/views.py [hunk 3]
```diff
@@ -39,110 +49,86 @@ def group_steem_golos(self, list_steem, list_golos):
...
+            except Exception as e:
```
Comment (AleksandrLeonov) [inline steepshot_io/graph/views.py]: Please handle JSONDecodeError and other requests error, like:

``` python
try:
    # some code
except JSONDecodeError as e:
    logger.error('Failed to parse json')
except ( ConnectionError, HTTPError) as e:
    logger.error('Failed to connect to....')
except Exception as e:
    logger.error('Other errors')
```
See the following docs for requests module: http://docs.python-requests.org/en/master/user/quickstart/#errors-and-exceptions

File: steepshot_io/graph/views.py [hunk 4]
```diff
@@ -39,110 +49,86 @@ def group_steem_golos(self, list_steem, list_golos):
...
-            values = self.get_active_users(platform=platform)
+            values = self._get_data(platform=platform, name_url='active_users', data_y='active_users', data_x='date_to')
         else:
-            values = self.get_active_users()
-        return render(request, self.template_name, {'values': values})
-
-
-class GetCountPostDaily(View):
-    template_name = 'count_post_daily.html'
+            values_steem, values_golos= self._get_data(name_url='active_users', data_y='active_users', data_x='date_to')
```
Comment (AleksandrLeonov) [inline steepshot_io/graph/views.py]: Please follow code style (PEP8).

File: steepshot_io/graph/views.py [hunk 5]
```diff
@@ -153,285 +139,273 @@ def group_steem_golos(self, list_steem, list_golos):
             idx += 1
         return res
 
-    def get_ratio_daily(self, platform=None):
-
+    def _get_data_dict(self, platform=None, reverse=True, data_x=None, data_y=None, name_url=None):
         if platform == 'steem':
```
Comment (AleksandrLeonov) [inline steepshot_io/graph/views.py]: The same. Please move this code to separate method.

File: steepshot_io/prod_settings.py [hunk 6]
```diff
@@ -83,9 +83,9 @@
                       'active_users': 'https://steepshot.org/api/v1/user/active/monthly',
                       'ration_daily': 'https://steepshot.org/api/v1/posts/ratio/daily',
                       'ration_monthly': 'https://steepshot.org/api/v1/posts/ratio/monthly',
-                      'new_users': 'https://qa.steepshot.org/api/v1/user/count/daily',
-                      'new_users_monthly': 'https://qa.steepshot.org/api/v1/user/count/unique',
-                      'users_percent_daily': 'https://qa.steepshot.org/api/v1/user/percent/unique',
+                      'new_users': 'https://steepshot.org/api/v1/user/count/daily',
```
Comment (alekseitylindus) [inline steepshot_io/prod_settings.py]: Please extract base URL to constants, for example:
``` python
STEEM_V1 = 'https://steepshot.org/api/v1'
STEEM_QA_V1 = 'https://qa.steepshot.org/api/v1'
GOLOS_V1 = 'https://golos.steepshot.org/api/v1'
GOLOS_QA_V1 = 'https://qa.golos.steepshot.org/api/v1'

...
REQUESTS_URL = {
    'new_users': '{url}/user/count/daily',
}
```

And then when you want to call this URL do something like this:
``` python
    res = requests.get(settings.REQUESTS_URL[name_url]).format(url=settings.STEEM_V1).json()  
    # but handle situations when there is no key 'name_url' in settings.REQUESTS_URL
```

File: steepshot_io/graph/views.py [hunk 7]
```diff
@@ -153,285 +139,273 @@ def group_steem_golos(self, list_steem, list_golos):
...
-            values = self.group_steem_golos(values_steem, values_golos)
-            flag_together = True
-            return render(request, self.template_name, {'values': values, 'flag': flag_together})
+            name_1 = 'count votes steem'
+            name_2 = 'count votes golos'
+            return render(request, self.template_name,
+                          {'values': values, 'flag': flag_together, 'name_1': name_1, 'name_2': name_2})
+
+
+# class AverageVotesWeekly(CountVotesWeekly):
```
Comment (alekseitylindus) [inline steepshot_io/graph/views.py:299]: Do we need this code in the future? If no - please delete

**Review by AleksandrLeonov — CHANGES_REQUESTED** [review]

**Review by alekseitylindus — CHANGES_REQUESTED** [review]

File: steepshot_io/graph/views.py [hunk 8]
```diff
@@ -1,32 +1,48 @@
...
+            values = self._get_data(name_key='count_fee', name_url='post_fee')
+        name = 'fee in day'
+        return render(request, self.template_name, {'values': values, 'name_1': name})
 
 
 class GetPostsCountMonthly(View):
-    template_name = 'count_posts.html'
+    template_name = 'graph.html'
 
     def group_steem_golos(self, list_steem, list_golos):
```
Comment (alekseitylindus) [inline steepshot_io/graph/views.py]: Maybe it's better to make more universal function?

For example this function can merge any count of arrays (but there is no checks to exclude IndexError):
``` python
    def group_steem_golos(self, *args):
        """
            Merge lists by first element of sublist and sum the second

            Example:
            l1 = [('2017-09-08', 10), ('2017-09-09', 12)]
            l2 = [('2017-09-08', 40), ('2017-09-09', 51)]
            result = [('2017-09-08', 50), ('2017-09-09', 63)]
        """
        return list(map(lambda x: (x[0][0], sum([y[1] for y in x])), zip(*args)))
```

File: steepshot_io/graph/views.py [hunk 9]
```diff
@@ -39,110 +55,72 @@ def group_steem_golos(self, list_steem, list_golos):
...
+            except JSONDecodeError as e:
+                logger.error('Failed to parse json {}'.format(e))
+            except (ConnectionError, HTTPError) as e:
+                logger.error('Failed to connect to {}'.format(e))
+            except Exception as e:
+                logger.error(e)
+            if reverse:
+                get_data.reverse()
+            res.append([(i[data_x], i[data_y]) for i in get_data])
+        values_total = [i for i in res]
```
Comment (alekseitylindus) [inline steepshot_io/graph/views.py]: Do we need to copy this list?

File: steepshot_io/graph/views.py [hunk 10]
```diff
@@ -3,7 +3,8 @@
 from django.shortcuts import render
 from django.views.generic import View
 from django.conf import settings
-
+from json.decoder import JSONDecodeError
+from requests.exceptions import HTTPError
```
Comment (AleksandrLeonov) [inline steepshot_io/graph/views.py]: Add import of 'ConnectionError' from requests module. ConnectionError which will raise from requests module will not handling.

File: steepshot_io/graph/views.py [hunk 11]
```diff
@@ -39,110 +55,72 @@ def group_steem_golos(self, list_steem, list_golos):
...
+            except Exception as e:
+                logger.error(e)
+            if reverse:
+                get_data.reverse()
+            res.append([(i[data_x], i[data_y]) for i in get_data])
+        values_total = [i for i in res]
+        return values_total
 
     def get(self, request):
         if 'platform' in request.GET:
```
Comment (alekseitylindus) [inline steepshot_io/graph/views.py]: If you will unify method `group_steem_golos` you can remove duplicate code and just write something like this:
``` python
    def get(self, request):
        values = self.group_steem_golos(
            *self._get_data(
                platform=request.GET.get('platform'),
                name_url='posts_count_monthly',
                data_x='date_to',
                data_y='posts_count'
            )
        )
        return render(request, self.template_name, {'values': values, 'name_1': 'counts posts'})
```

**Review by AleksandrLeonov — CHANGES_REQUESTED** [review]

**Review by alekseitylindus — CHANGES_REQUESTED** [review]

File: steepshot_io/graph/views.py (hunk 8 above)
Comment (roman4hik) [reply steepshot_io/graph/views.py]: your function do not work

File: steepshot_io/graph/views.py (hunk 7 above)
Comment (roman4hik) [reply steepshot_io/graph/views.py:299]: this code need in future

File: steepshot_io/graph/views.py (hunk 8 above)
Comment (alekseitylindus) [reply steepshot_io/graph/views.py]: It works well, you copied class method and tried to call it like a separate function. You didn't remove 'self' argument from arguments.
